Improve joint simulation

nickswalker opened the pull request:
Add gazebo_ros_control PID gains
Tuned manually to make movement via the teleop interface behave well
Wheels still unspecified because they cause the robot to collapse
Use effort joint interface for the gripper so that gripper contacts are well simulated

I have been able to pick up objects using these gains and a simulation step size of .0005. I still notice that the lift joint is sluggish on the way up and comes down with an implausible amount of force, which can cause the base to shake and shift position noticeably. It's possible that better tuning of the controller parameters will help, but it may also be the case that the simulated robot geometry/inertias need to be tweaked.

Closes #39

nickswalker committed:
Improve joint simulation

Add gazebo_ros_control PID gains
Tuned manually to make movement via the teleop interface behave well
Wheels still unspecified because they cause the robot to collapse
Use effort joint interface for the gripper so that gripper contacts are well simulated

vatanaksoytezer reviewed (commented):
@nickswalker thank you for the PR, this looks great! So, I've been testing this and happy with the gains so far. You are right about inertias as well, I don't know if you have hit that yet but if you try to grab a coke can or rather any object heavier than 200 grams when the arm is fully extended, the robot will fall over.

Having said that these gains do not play well with "default gazebo worlds" which has a step size of `0.001`. I would prefer to tweak the gains a little more to ensure it plays well with default step size. I will look at what plays well, but if you have some gains tested if you can tweak it to include these number I will be more than happy to merge it.

nickswalker committed:
Use effort control for all arm joints

Removes most of the instability in simulation behavior when lifting objects

nickswalker committed:
Increase mass of base

Makes tipping behavior in simulation more plausible
I assume that the mass of an assembled base is at least this heavy, given that the robot is around 50lbs total

nickswalker commented:
The instability seems to have been from the arm direct position simulation. Here's [the behavior with that controller using effort instead](https://drive.google.com/file/d/1iho9VjNn6e1xSm-6_hqusqJD3dUWf9-Z/view?usp=sharing). I increased the mass of the base to 6kg to get the tipping behavior to look reasonable, but it looks like most of the robot's mass is not accounted for in the URDF. It'd be best if someone could measure this with a partially disassembled robot.
